test/add.js

```diff
@@ -38,222 +38,268 @@ function makeFileWithInlineSourceMap() {
-		t.ok(File.isVinyl(data), 'should return a vinyl file');
-		t.ok(data.sourceMap, 'should add a source map object');
-		t.equal(String(data.sourceMap.version), '3', 'should have version 3');
-		t.equal(data.sourceMap.sources[0], 'helloworld.js', 'should add file to sources');
-		t.equal(data.sourceMap.sourcesContent[0], sourceContent, 'should add file content to sourcesContent');
-		t.deepEqual(data.sourceMap.names, [], 'should add empty names');
-		t.equal(data.sourceMap.mappings, '', 'should add empty mappings');
-		t.end();
+	it('should not accept a stream as argument', function(done) {
+		[
+			{ type: 'a stream', val: new stream.Readable() }
+		].map(function(obj) {
+			sourcemaps.add(obj.val, function(err, file) {
+				expect(err instanceof Error && err.message === 'vinyl-sourcemap-add: Not a vinyl file').toExist();
+				done();
+			});
+		});
 	});
-});
 
-test('add: should add a valid sourcemap if wished', function (t) {
-	sourcemaps.add(makeFile(), { identityMap: true }, function(err, data) {
-		t.ok(data, 'should pass something through');
-		t.ok(data instanceof File, 'should pass a vinyl file through');
-		t.ok(data.sourceMap, 'should add a source map object');
-		t.equal(String(data.sourceMap.version), '3', 'should have version 3');
-		t.equal(data.sourceMap.sources[0], 'helloworld.js', 'should add file to sources');
-		t.equal(data.sourceMap.sourcesContent[0], sourceContent, 'should add file content to sourcesContent');
-		t.deepEqual(data.sourceMap.names, ['helloWorld', 'console', 'log'], 'should add correct names');
-		t.equal(data.sourceMap.mappings, 'AAAA,YAAY;;AAEZ,SAASA,UAAU,CAAC,EAAE;CACrBC,OAAO,CAACC,GAAG,CAAC,cAAc,CAAC;AAC5B', 'should add correct mappings');
-		t.end();
-	});
-});
+		it('should not accept undefined as options argument', function(done) {
+			var file = makeFile();
+			[
+				{ type: 'undefined' }
+			].map(function(obj) {
+				sourcemaps.add(file, obj.val, function(err, file) {
+					expect(err instanceof Error && err.message === 'vinyl-sourcemap-add: Invalid argument: options').toExist();
+					done();
+				});
+			});
+		});
 
-test('add: should add a valid source map for CSS if wished', function (t) {
-	sourcemaps.add(makeFileCSS(), { identityMap: true }, function(err, data) {
-		t.ok(data, 'should pass something through');
-		t.ok(data instanceof File, 'should pass a vinyl file through');
-		t.ok(data.sourceMap, 'should add a source map object');
-		t.equal(String(data.sourceMap.version), '3', 'should have version 3');
-		t.equal(data.sourceMap.sources[0], 'test.css', 'should add file to sources');
-		t.equal(data.sourceMap.sourcesContent[0], sourceContentCSS, 'should add file content to sourcesContent');
-		t.deepEqual(data.sourceMap.names, [], 'should add correct names');
-		t.equal(data.sourceMap.mappings, 'CAAC;EACC;EACA', 'should add correct mappings');
-		t.end();
-	});
-});
+		it('should not accept null as options argument', function(done) {
+			var file = makeFile();
+			[
```

Comment:
No need to use an array and `.map` here. You can just replace `obj.val` with the value you are testing